geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/util/RasterUtils.scala

```diff
@@ -0,0 +1,305 @@
+        //TODO: check for corner cases: https://geomesa.atlassian.net/browse/GEOMESA-758
+        val mosaicX = Math.round(queryEnv.getSpan(0) / accumuloRasterXRes).toInt
+        val mosaicY = Math.round(queryEnv.getSpan(1) / accumuloRasterYRes).toInt
+        if (mosaicX <= 0 || mosaicY <= 0) {
+          (null, 1)
+        } else {
+          var count = 1
+          val mosaic = allocateBufferedImage(mosaicX, mosaicY, firstRaster.chunk)
+          writeToMosaic(mosaic, firstRaster, queryEnv, accumuloRasterXRes, accumuloRasterYRes)
+          while (chunks.hasNext) {
+            writeToMosaic(mosaic, chunks.next(), queryEnv, accumuloRasterXRes, accumuloRasterYRes)
+            count += 1
+          }
+          (scaleBufferedImage(queryWidth, queryHeight, mosaic), count)
+        }
+      }
+    }
+  }
+
+  def scaleBufferedImage(newWidth: Int, newHeight: Int, image: BufferedImage): BufferedImage = {
+    if (image.getWidth == newWidth && image.getHeight == newHeight) {
+      image
+    } else {
+      if (newWidth < 1 || newHeight < 1) null
+      else {
+        val result = allocateBufferedImage(newWidth, newHeight, image)
+        val resGraphics = result.createGraphics()
+        resGraphics.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR)
+        resGraphics.drawImage(image, 0, 0, newWidth, newHeight, null)
+        resGraphics.dispose()
+        result
+      }
+    }
+  }
+
+  def cropRaster(raster: Raster, cropEnv: Envelope): Option[BufferedImage] = {
+    val rasterEnv = raster.referencedEnvelope
+    val intersection = rasterEnv.intersection(envelopeToReferencedEnvelope(cropEnv))
+    if (intersection.equals(rasterEnv)) {
+      Some(renderedImageToBufferedImage(raster.chunk))
+    } else {
+      val chunkXRes = rasterEnv.getWidth / raster.chunk.getWidth
+      val chunkYRes = rasterEnv.getHeight / raster.chunk.getHeight
+      //TODO: check for corner cases: https://geomesa.atlassian.net/browse/GEOMESA-758
+      val widthP  = Math.round(intersection.getWidth / chunkXRes)
+      val heightP = Math.round(intersection.getHeight / chunkYRes)
+      if (widthP > 0 && heightP > 0) {
+        val uLX = Math.max(Math.floor((intersection.getMinX - rasterEnv.getMinimum(0)) / chunkXRes).toInt, 0)
+        val uLY = Math.max(Math.floor((rasterEnv.getMaximum(1) - intersection.getMaxY) / chunkYRes).toInt, 0)
+        val wTemp = Math.max(Math.ceil(intersection.getWidth / chunkXRes).toInt, 0)
+        val w = if (wTemp + uLX > raster.chunk.getWidth) {
+          raster.chunk.getWidth - uLX
+        } else {
+          wTemp
+        }
+        val hTemp = Math.max(Math.ceil(intersection.getHeight / chunkYRes).toInt, 0)
+        val h = if (hTemp + uLY > raster.chunk.getHeight) {
+          raster.chunk.getHeight - uLY
+        } else {
+          hTemp
+        }
+        val b = renderedImageToBufferedImage(raster.chunk)
+        Some(bufferCrop(b, uLX, uLY, w, h))
+      } else None
```

Comment:
nope, there is another if statement where I check to see if the area of the crop is at least 1x1 pixels. I will put some comments explaining what is going on with some clearer variable names.